pandas/tests/arrays/test_datetimelike.py

```diff
@@ -1,13 +1,50 @@
 # -*- coding: utf-8 -*-
 import numpy as np
+import pytest
 
 import pandas as pd
+import pandas.util.testing as tm
 
 from pandas.core.arrays.datetimes import DatetimeArrayMixin
 from pandas.core.arrays.timedeltas import TimedeltaArrayMixin
 from pandas.core.arrays.period import PeriodArrayMixin
 
 
+# TODO: more freq variants
+@pytest.fixture(params=['D', 'B', 'W', 'M', 'Q', 'Y'])
+def period_index(request):
+    """
+    A fixture to provide PeriodIndex objects with different frequencies.
+
+    Most PeriodArray behavior is already tested in PeriodIndex tests,
+    so here we just test that the PeriodArray behavior matches
+    the PeriodIndex behavior.
+    """
+    freqstr = request.param
+    # TODO: non-monotone indexes; NaTs, different start dates
+    pi = pd.period_range(start=pd.Timestamp('2000-01-01'),
```

Comment:
Hmm, sorry forgot to mention this earlier. Is it possible to include a name on these? It may complicate the tests a bit.

Or maybe just double check that we have other tests ensure that the index `.to_timestamp` and `.to_period` assert the name is passed through.

Reply:
> Is it possible to include a name on these?

Sure I guess.  My thought for both of these fixtures is to try to do something hypothesis-like to cover the whole space of Period/DatetimeIndexes that might occur in the wild.

> assert the name is passed through.

... but it isn't.  The Array classes don't have a `name`.  Or am I not understanding the suggestion?

Reply:
I probably wasn't being clear, and I think I misunderstood the eventual goal of these tests. This looks fine, and I've verified that we do have other tests for `PeriodIndex.to_timestmap`'s name.